###  데이터 불러오기

In [1]:
import pickle
import gzip
import pandas as pd

In [3]:
# col 생략 없이 출력
pd.set_option('display.max_columns', None)
pd.set_option('display.max_colwidth', -1)

C:\Users\elyse\AppData\Local\Temp\ipykernel_12508\94279240.py:3: FutureWarning: Passing a negative integer is deprecated in version 1.0 and will not be supported in future version. Instead, use None to not limit the column width.
  pd.set_option('display.max_colwidth', -1)


In [2]:
with gzip.open('땡큐스티커.pickle','rb') as f:
    data = pickle.load(f)
df_sticker = pd.DataFrame(data)

with gzip.open('접속횟수.pickle','rb') as f:
    data = pickle.load(f)
df_access = pd.DataFrame(data)

with gzip.open('채팅팬.pickle','rb') as f:
    data = pickle.load(f)
df_fan = pd.DataFrame(data)

with gzip.open('채팅횟수.pickle','rb') as f:
    data = pickle.load(f)
df_chat= pd.DataFrame(data)

with gzip.open('celeb_list.pickle','rb') as f:
    data = pickle.load(f)
df_list = pd.DataFrame(data)



In [5]:
df_sticker.head(10)

chat_room_id  \
0  sendbird_group_channel_51673049_e9127466665f7535749033f986c3e01c7a67cfc1   
1  sendbird_group_channel_82337532_c81ea37759a4761f4f6989e5f0f170f2e7a723fd   
2  sendbird_group_channel_82333993_7fe5b9c11092fdc107a23a6dcb482634eb75b3da   
3  sendbird_group_channel_90863793_15d27a6ad0ea225f4547c61b2b6f7fd58989e4b7   
4  sendbird_group_channel_90683697_dc65fb5afa9b3c6dfae1b4cf4361c433a796d7e5   
5  sendbird_group_channel_90863793_15d27a6ad0ea225f4547c61b2b6f7fd58989e4b7   
6  sendbird_group_channel_90095139_b2c40556db1588f9acc22ae0dc0f2228860a1ccc   
7  sendbird_group_channel_90683697_dc65fb5afa9b3c6dfae1b4cf4361c433a796d7e5   
8  sendbird_group_channel_90095139_b2c40556db1588f9acc22ae0dc0f2228860a1ccc   
9  sendbird_group_channel_82337532_c81ea37759a4761f4f6989e5f0f170f2e7a723fd   

  user_nickname chat_celeb_nickname product_price_string  \
0  밝은달           🍒체리쉬_문백찬            30                    
1  누리            🍒체리쉬_반이안            10                    
2  수피아           🍒체리쉬_윤호림            30                    
3  장미팔           유 앤🌸                20                    
4  에엘            예꿍_                 100                   
5  장미팔           유 앤🌸                35                    
6  갱리            나는야꼬등어              10                    
7  겨울            예꿍_                 10                    
8  shin-ki       나는야꼬등어              35                    
9  풰이            🍒체리쉬_반이안            30                    

                       time  
0 2022-09-04 00:10:31+00:00  
1 2022-09-04 00:19:52+00:00  
2 2022-09-04 00:39:44+00:00  
3 2022-09-04 10:18:29+00:00  
4 2022-09-04 13:16:43+00:00  
5 2022-09-04 13:30:29+00:00  
6 2022-09-04 14:05:17+00:00  
7 2022-09-04 14:09:07+00:00  
8 2022-09-04 14:11:06+00:00  
9 2022-09-04 16:14:39+00:00

In [6]:
df_sticker.chat_room_id.nunique()

195

In [7]:
df_access.chat_room_id.nunique()

1106

In [8]:
df_access.head(10)

chat_room_id  \
0  sendbird_group_channel_51673049_e9127466665f7535749033f986c3e01c7a67cfc1   
1  sendbird_group_channel_51673049_e9127466665f7535749033f986c3e01c7a67cfc1   
2  sendbird_group_channel_90584763_e7b41d72ca4a6efdef2d11c6c7efd4a9e9ed57a8   
3  sendbird_group_channel_90584763_e7b41d72ca4a6efdef2d11c6c7efd4a9e9ed57a8   
4  sendbird_group_channel_90584763_e7b41d72ca4a6efdef2d11c6c7efd4a9e9ed57a8   
5  sendbird_group_channel_90584763_e7b41d72ca4a6efdef2d11c6c7efd4a9e9ed57a8   
6  sendbird_group_channel_90584763_e7b41d72ca4a6efdef2d11c6c7efd4a9e9ed57a8   
7  sendbird_group_channel_90584763_e7b41d72ca4a6efdef2d11c6c7efd4a9e9ed57a8   
8  sendbird_group_channel_90584763_e7b41d72ca4a6efdef2d11c6c7efd4a9e9ed57a8   
9  sendbird_group_channel_90584763_e7b41d72ca4a6efdef2d11c6c7efd4a9e9ed57a8   

  user_nickname                      time user_type     mp_event_name  
0  🍒체리쉬_문백찬     2022-09-04 00:00:00+00:00  CELEB     chat_list_viewed  
1  🍒체리쉬_문백찬     2022-09-04 00:00:00+00:00  CELEB     chat_list_viewed  
2  정다별이         2022-09-04 00:00:03+00:00  CELEB     chat_list_viewed  
3  정다별이         2022-09-04 00:00:03+00:00  CELEB     chat_txt_sent     
4  정다별이         2022-09-04 00:00:08+00:00  CELEB     chat_txt_sent     
5  정다별이         2022-09-04 00:00:08+00:00  CELEB     chat_list_viewed  
6  정다별이         2022-09-04 00:00:13+00:00  CELEB     chat_list_viewed  
7  정다별이         2022-09-04 00:00:13+00:00  CELEB     chat_txt_sent     
8  정다별이         2022-09-04 00:00:17+00:00  CELEB     chat_list_viewed  
9  정다별이         2022-09-04 00:00:17+00:00  CELEB     chat_txt_sent

In [9]:
df_access.mp_event_name.unique()

array(['chat_list_viewed', 'chat_txt_sent', '$ae_session', 'Page View',
       'celeb_list_viewed', 'Entry Page\tentry page viewed',
       'setting_page_viewed', 'celeb_profile_page_viewed', 'chat_isFirst',
       'chat_page_btnIcons_click', '$identify', 'login succeeded',
       'login_page_view', 'into_loginPage', 'logout_succeed',
       'login_btn_click', '$ae_updated', 'chat_button_clicked',
       'registration_btn_click', 'buy_a_shot', 'share_chatLink_clicked',
       'buy_a_shot_complete', '$ae_iap', 'Product Click',
       'send_thankU_sticker', 'chat_leave', 'saved_thankU_sticker',
       'joinus succeeded', '$ae_crashed', '$create_alias',
       'chat_pressedBtn_click'], dtype=object)

In [10]:
df_access.head()

chat_room_id  \
0  sendbird_group_channel_51673049_e9127466665f7535749033f986c3e01c7a67cfc1   
1  sendbird_group_channel_51673049_e9127466665f7535749033f986c3e01c7a67cfc1   
2  sendbird_group_channel_90584763_e7b41d72ca4a6efdef2d11c6c7efd4a9e9ed57a8   
3  sendbird_group_channel_90584763_e7b41d72ca4a6efdef2d11c6c7efd4a9e9ed57a8   
4  sendbird_group_channel_90584763_e7b41d72ca4a6efdef2d11c6c7efd4a9e9ed57a8   

  user_nickname                      time user_type     mp_event_name  
0  🍒체리쉬_문백찬     2022-09-04 00:00:00+00:00  CELEB     chat_list_viewed  
1  🍒체리쉬_문백찬     2022-09-04 00:00:00+00:00  CELEB     chat_list_viewed  
2  정다별이         2022-09-04 00:00:03+00:00  CELEB     chat_list_viewed  
3  정다별이         2022-09-04 00:00:03+00:00  CELEB     chat_txt_sent     
4  정다별이         2022-09-04 00:00:08+00:00  CELEB     chat_txt_sent

In [11]:
df_fan.head()

chat_room_id  \
0  sendbird_group_channel_51673049_e9127466665f7535749033f986c3e01c7a67cfc1   
1  sendbird_group_channel_90584763_e7b41d72ca4a6efdef2d11c6c7efd4a9e9ed57a8   
2  sendbird_group_channel_51673049_e9127466665f7535749033f986c3e01c7a67cfc1   
3  sendbird_group_channel_85027934_96d14caeb215fa030b76ab8a449dafae0e252bfe   
4  sendbird_group_channel_51673049_e9127466665f7535749033f986c3e01c7a67cfc1   

  user_nickname user_type                      time  
0  여니            FAN      2022-09-04 00:00:18+00:00  
1  포로리짱          FAN      2022-09-04 00:00:20+00:00  
2  여니            FAN      2022-09-04 00:00:25+00:00  
3  김물든           FAN      2022-09-04 00:00:27+00:00  
4  여니            FAN      2022-09-04 00:00:34+00:00

In [12]:
df_fan.chat_room_id.nunique()

614

In [13]:
df_chat.head()

chat_room_id  \
0  sendbird_group_channel_90584763_e7b41d72ca4a6efdef2d11c6c7efd4a9e9ed57a8   
1  sendbird_group_channel_90584763_e7b41d72ca4a6efdef2d11c6c7efd4a9e9ed57a8   
2  sendbird_group_channel_90584763_e7b41d72ca4a6efdef2d11c6c7efd4a9e9ed57a8   
3  sendbird_group_channel_90584763_e7b41d72ca4a6efdef2d11c6c7efd4a9e9ed57a8   
4  sendbird_group_channel_90584763_e7b41d72ca4a6efdef2d11c6c7efd4a9e9ed57a8   

  user_nickname  message_length                      time  
0  정다별이          1.0            2022-09-04 00:00:03+00:00  
1  정다별이          9.0            2022-09-04 00:00:08+00:00  
2  정다별이          10.0           2022-09-04 00:00:13+00:00  
3  정다별이          12.0           2022-09-04 00:00:17+00:00  
4  정다별이          5.0            2022-09-04 00:00:18+00:00

In [14]:
df_chat.chat_room_id.nunique()

819

In [15]:
df_list.head()

user_nickname  \
0  소나무            
1  아라7            
2  밤양갱            
3  Yewon          
4  mcel01         

                                                               chat_room_id  
0  sendbird_group_channel_85249736_52d2fa2d3517ffc822f7964bdf82c80e801033cd  
1  sendbird_group_channel_89939017_5d92753dd838235ebee2617f3446a6a8707fc3c5  
2  sendbird_group_channel_95193234_fe13fff1331cae16e336ef3b5968f23e59a3d01c  
3  sendbird_group_channel_92986328_49d9846cdbac6b1d63b58ba560ff895913f9b64e  
4  sendbird_group_channel_66268842_b0e38258ae02c78be067e9abb11ea924d1887994

In [16]:
df_list.shape

(883, 2)

### df_fan에서 채팅최초일자, 채팅최근일자, 채팅기간(day/week), 누적채팅량, 최근 일자기준 1주일간 채팅량 구하기

In [28]:
df_fan

chat_room_id  \
0        sendbird_group_channel_51673049_e9127466665f7535749033f986c3e01c7a67cfc1    
1        sendbird_group_channel_90584763_e7b41d72ca4a6efdef2d11c6c7efd4a9e9ed57a8    
2        sendbird_group_channel_51673049_e9127466665f7535749033f986c3e01c7a67cfc1    
3        sendbird_group_channel_85027934_96d14caeb215fa030b76ab8a449dafae0e252bfe    
4        sendbird_group_channel_51673049_e9127466665f7535749033f986c3e01c7a67cfc1    
...                                                                           ...    
1055946  sendbird_group_channel_94070327_17ef4e05fa7263db35e76a9fc66b83ba3bc90c17    
1055947  sendbird_group_channel_49516685_8a1efc458554e7fbdcee85684a1384965bd0d31f    
1055948  sendbird_group_channel_94516740_608949752e067685ec73cfa3491f2b90ab79c542    
1055949  sendbird_group_channel_102487442_7c9f52e0fcbe0b8d65884c770327899ffd2edb3b   
1055950  sendbird_group_channel_92280528_bb1148216c54f7b138d6798262da032358a4f845    

        user_nickname user_type                      time  
0        여니            FAN      2022-09-04 00:00:18+00:00  
1        포로리짱          FAN      2022-09-04 00:00:20+00:00  
2        여니            FAN      2022-09-04 00:00:25+00:00  
3        김물든           FAN      2022-09-04 00:00:27+00:00  
4        여니            FAN      2022-09-04 00:00:34+00:00  
...      ..            ...                            ...  
1055946  온오프           FAN      2023-01-20 14:58:10+00:00  
1055947  zero_won      FAN      2023-01-20 14:58:17+00:00  
1055948  T_L           FAN      2023-01-20 14:58:18+00:00  
1055949  892           FAN      2023-01-20 14:58:19+00:00  
1055950  뇡면            FAN      2023-01-20 14:59:56+00:00  

[1055951 rows x 4 columns]

In [130]:
#df_tmp1 채팅방 별 가장 최근 채팅일자
df_tmp1 = df_fan.groupby('chat_room_id',as_index=False).max()[['chat_room_id','time']]
#날짜만 가져오기
df_tmp1['time'] = df_tmp1['time'].apply(lambda x :x.date)
#컬럼이름을 last_chat_date로 변경
df_tmp1.rename(columns = {'time':'last_fan_chat_date'}, inplace=True)
df_tmp1

chat_room_id  \
0    sendbird_group_channel_100098601_8588e70ceb6ad4538f8cd750040503f0281e2256   
1    sendbird_group_channel_100230212_d09bcba2f742f5e2eccfa93929cac768ea781690   
2    sendbird_group_channel_100311155_0b3a8d43c52f47f7fc8a64f7be914c5449f3db9e   
3    sendbird_group_channel_100349582_b3e0ea76ba8e8e82d84c7883fb00249a381b0a09   
4    sendbird_group_channel_100393628_3311cc39eeafde271bc8e3c2c74b6c3b6abb9407   
..                                                                         ...   
609  sendbird_group_channel_99795834_500b186865d605408b22e8774935cb2b7decbd86    
610  sendbird_group_channel_99850650_d7bd96328f34a17ae3e1d754f234f68a17304ce0    
611  sendbird_group_channel_99874745_64724648d72307db2142bc58979ea6e33e09855a    
612  sendbird_group_channel_99875351_3591638d2b87c2551d3521609dd4c3c19fae0d4b    
613  sendbird_group_channel_99954388_1b9a4a36f692e3b4f9b39c0f520e43a498f6f12e    

    last_fan_chat_date  
0    2022-11-30         
1    2023-01-20         
2    2023-01-01         
3    2023-01-20         
4    2023-01-04         
..          ...         
609  2022-11-13         
610  2022-11-19         
611  2022-11-12         
612  2023-01-20         
613  2022-11-13         

[614 rows x 2 columns]

In [131]:
#df_tmp2 채팅방 별 가장 최초 채팅일자(기간중)
df_tmp2 = df_fan.groupby('chat_room_id',as_index=False).min()[['chat_room_id','time']]
#날짜만 가져오기
df_tmp2['time'] = df_tmp2['time'].apply(lambda x :x.date)
#컬럼이름을 last_chat_date로 변경
df_tmp2.rename(columns = {'time':'first_fan_chat_date'}, inplace=True)
df_tmp2

chat_room_id  \
0    sendbird_group_channel_100098601_8588e70ceb6ad4538f8cd750040503f0281e2256   
1    sendbird_group_channel_100230212_d09bcba2f742f5e2eccfa93929cac768ea781690   
2    sendbird_group_channel_100311155_0b3a8d43c52f47f7fc8a64f7be914c5449f3db9e   
3    sendbird_group_channel_100349582_b3e0ea76ba8e8e82d84c7883fb00249a381b0a09   
4    sendbird_group_channel_100393628_3311cc39eeafde271bc8e3c2c74b6c3b6abb9407   
..                                                                         ...   
609  sendbird_group_channel_99795834_500b186865d605408b22e8774935cb2b7decbd86    
610  sendbird_group_channel_99850650_d7bd96328f34a17ae3e1d754f234f68a17304ce0    
611  sendbird_group_channel_99874745_64724648d72307db2142bc58979ea6e33e09855a    
612  sendbird_group_channel_99875351_3591638d2b87c2551d3521609dd4c3c19fae0d4b    
613  sendbird_group_channel_99954388_1b9a4a36f692e3b4f9b39c0f520e43a498f6f12e    

    first_fan_chat_date  
0    2022-11-15          
1    2022-11-15          
2    2022-11-16          
3    2022-11-17          
4    2022-12-06          
..          ...          
609  2022-11-13          
610  2022-11-12          
611  2022-11-12          
612  2022-11-12          
613  2022-11-13          

[614 rows x 2 columns]

In [132]:
#df_tmp3 채팅방 팬 채팅 누적 횟수
df_tmp3 = df_fan.groupby('chat_room_id',as_index=False).count()[['chat_room_id','time']]
#컬럼이름을 chat_count로 변경
df_tmp3.rename(columns = {'time':'fan_chat_count'}, inplace=True)
df_tmp3

chat_room_id  \
0    sendbird_group_channel_100098601_8588e70ceb6ad4538f8cd750040503f0281e2256   
1    sendbird_group_channel_100230212_d09bcba2f742f5e2eccfa93929cac768ea781690   
2    sendbird_group_channel_100311155_0b3a8d43c52f47f7fc8a64f7be914c5449f3db9e   
3    sendbird_group_channel_100349582_b3e0ea76ba8e8e82d84c7883fb00249a381b0a09   
4    sendbird_group_channel_100393628_3311cc39eeafde271bc8e3c2c74b6c3b6abb9407   
..                                                                         ...   
609  sendbird_group_channel_99795834_500b186865d605408b22e8774935cb2b7decbd86    
610  sendbird_group_channel_99850650_d7bd96328f34a17ae3e1d754f234f68a17304ce0    
611  sendbird_group_channel_99874745_64724648d72307db2142bc58979ea6e33e09855a    
612  sendbird_group_channel_99875351_3591638d2b87c2551d3521609dd4c3c19fae0d4b    
613  sendbird_group_channel_99954388_1b9a4a36f692e3b4f9b39c0f520e43a498f6f12e    

     fan_chat_count  
0    23              
1    891             
2    22              
3    1106            
4    4               
..  ..               
609  1               
610  11              
611  7               
612  1988            
613  5               

[614 rows x 2 columns]

In [133]:
df_tmp1 = df_tmp1.merge(df_tmp2,on='chat_room_id')
df_tmp1['fan_chat_period']=df_tmp1['last_fan_chat_date']-df_tmp1['first_fan_chat_date']
df_tmp1['fan_chat_period'] = df_tmp1['fan_chat_period'].dt.days+1
df_tmp1

chat_room_id  \
0    sendbird_group_channel_100098601_8588e70ceb6ad4538f8cd750040503f0281e2256   
1    sendbird_group_channel_100230212_d09bcba2f742f5e2eccfa93929cac768ea781690   
2    sendbird_group_channel_100311155_0b3a8d43c52f47f7fc8a64f7be914c5449f3db9e   
3    sendbird_group_channel_100349582_b3e0ea76ba8e8e82d84c7883fb00249a381b0a09   
4    sendbird_group_channel_100393628_3311cc39eeafde271bc8e3c2c74b6c3b6abb9407   
..                                                                         ...   
609  sendbird_group_channel_99795834_500b186865d605408b22e8774935cb2b7decbd86    
610  sendbird_group_channel_99850650_d7bd96328f34a17ae3e1d754f234f68a17304ce0    
611  sendbird_group_channel_99874745_64724648d72307db2142bc58979ea6e33e09855a    
612  sendbird_group_channel_99875351_3591638d2b87c2551d3521609dd4c3c19fae0d4b    
613  sendbird_group_channel_99954388_1b9a4a36f692e3b4f9b39c0f520e43a498f6f12e    

    last_fan_chat_date first_fan_chat_date  fan_chat_period  
0    2022-11-30         2022-11-15          16               
1    2023-01-20         2022-11-15          67               
2    2023-01-01         2022-11-16          47               
3    2023-01-20         2022-11-17          65               
4    2023-01-04         2022-12-06          30               
..          ...                ...          ..               
609  2022-11-13         2022-11-13          1                
610  2022-11-19         2022-11-12          8                
611  2022-11-12         2022-11-12          1                
612  2023-01-20         2022-11-12          70               
613  2022-11-13         2022-11-13          1                

[614 rows x 4 columns]

In [134]:
#날짜로 몇주인지 구하는 함수
def div_week(num):
    if num <6:
        return 1
    elif num % 7 == 0:
        return int(num / 7)
    else:
        return int(num // 7+1)

In [135]:
df_tmp1['fan_chat_week'] = df_tmp1['fan_chat_period'].apply(div_week)
df_tmp1

chat_room_id  \
0    sendbird_group_channel_100098601_8588e70ceb6ad4538f8cd750040503f0281e2256   
1    sendbird_group_channel_100230212_d09bcba2f742f5e2eccfa93929cac768ea781690   
2    sendbird_group_channel_100311155_0b3a8d43c52f47f7fc8a64f7be914c5449f3db9e   
3    sendbird_group_channel_100349582_b3e0ea76ba8e8e82d84c7883fb00249a381b0a09   
4    sendbird_group_channel_100393628_3311cc39eeafde271bc8e3c2c74b6c3b6abb9407   
..                                                                         ...   
609  sendbird_group_channel_99795834_500b186865d605408b22e8774935cb2b7decbd86    
610  sendbird_group_channel_99850650_d7bd96328f34a17ae3e1d754f234f68a17304ce0    
611  sendbird_group_channel_99874745_64724648d72307db2142bc58979ea6e33e09855a    
612  sendbird_group_channel_99875351_3591638d2b87c2551d3521609dd4c3c19fae0d4b    
613  sendbird_group_channel_99954388_1b9a4a36f692e3b4f9b39c0f520e43a498f6f12e    

    last_fan_chat_date first_fan_chat_date  fan_chat_period  fan_chat_week  
0    2022-11-30         2022-11-15          16               3              
1    2023-01-20         2022-11-15          67               10             
2    2023-01-01         2022-11-16          47               7              
3    2023-01-20         2022-11-17          65               10             
4    2023-01-04         2022-12-06          30               5              
..          ...                ...          ..              ..              
609  2022-11-13         2022-11-13          1                1              
610  2022-11-19         2022-11-12          8                2              
611  2022-11-12         2022-11-12          1                1              
612  2023-01-20         2022-11-12          70               10             
613  2022-11-13         2022-11-13          1                1              

[614 rows x 5 columns]

In [136]:
#df_tmp3도 합침
df_tmp1 = df_tmp1.merge(df_tmp3,on='chat_room_id')
df_tmp1

chat_room_id  \
0    sendbird_group_channel_100098601_8588e70ceb6ad4538f8cd750040503f0281e2256   
1    sendbird_group_channel_100230212_d09bcba2f742f5e2eccfa93929cac768ea781690   
2    sendbird_group_channel_100311155_0b3a8d43c52f47f7fc8a64f7be914c5449f3db9e   
3    sendbird_group_channel_100349582_b3e0ea76ba8e8e82d84c7883fb00249a381b0a09   
4    sendbird_group_channel_100393628_3311cc39eeafde271bc8e3c2c74b6c3b6abb9407   
..                                                                         ...   
609  sendbird_group_channel_99795834_500b186865d605408b22e8774935cb2b7decbd86    
610  sendbird_group_channel_99850650_d7bd96328f34a17ae3e1d754f234f68a17304ce0    
611  sendbird_group_channel_99874745_64724648d72307db2142bc58979ea6e33e09855a    
612  sendbird_group_channel_99875351_3591638d2b87c2551d3521609dd4c3c19fae0d4b    
613  sendbird_group_channel_99954388_1b9a4a36f692e3b4f9b39c0f520e43a498f6f12e    

    last_fan_chat_date first_fan_chat_date  fan_chat_period  fan_chat_week  \
0    2022-11-30         2022-11-15          16               3               
1    2023-01-20         2022-11-15          67               10              
2    2023-01-01         2022-11-16          47               7               
3    2023-01-20         2022-11-17          65               10              
4    2023-01-04         2022-12-06          30               5               
..          ...                ...          ..              ..               
609  2022-11-13         2022-11-13          1                1               
610  2022-11-19         2022-11-12          8                2               
611  2022-11-12         2022-11-12          1                1               
612  2023-01-20         2022-11-12          70               10              
613  2022-11-13         2022-11-13          1                1               

     fan_chat_count  
0    23              
1    891             
2    22              
3    1106            
4    4               
..  ..               
609  1               
610  11              
611  7               
612  1988            
613  5               

[614 rows x 6 columns]

In [137]:
df_tmp1['avg_fan_chat'] = round ( df_tmp1['fan_chat_count'] /  df_tmp1['fan_chat_week'] , 2)
df_tmp1

chat_room_id  \
0    sendbird_group_channel_100098601_8588e70ceb6ad4538f8cd750040503f0281e2256   
1    sendbird_group_channel_100230212_d09bcba2f742f5e2eccfa93929cac768ea781690   
2    sendbird_group_channel_100311155_0b3a8d43c52f47f7fc8a64f7be914c5449f3db9e   
3    sendbird_group_channel_100349582_b3e0ea76ba8e8e82d84c7883fb00249a381b0a09   
4    sendbird_group_channel_100393628_3311cc39eeafde271bc8e3c2c74b6c3b6abb9407   
..                                                                         ...   
609  sendbird_group_channel_99795834_500b186865d605408b22e8774935cb2b7decbd86    
610  sendbird_group_channel_99850650_d7bd96328f34a17ae3e1d754f234f68a17304ce0    
611  sendbird_group_channel_99874745_64724648d72307db2142bc58979ea6e33e09855a    
612  sendbird_group_channel_99875351_3591638d2b87c2551d3521609dd4c3c19fae0d4b    
613  sendbird_group_channel_99954388_1b9a4a36f692e3b4f9b39c0f520e43a498f6f12e    

    last_fan_chat_date first_fan_chat_date  fan_chat_period  fan_chat_week  \
0    2022-11-30         2022-11-15          16               3               
1    2023-01-20         2022-11-15          67               10              
2    2023-01-01         2022-11-16          47               7               
3    2023-01-20         2022-11-17          65               10              
4    2023-01-04         2022-12-06          30               5               
..          ...                ...          ..              ..               
609  2022-11-13         2022-11-13          1                1               
610  2022-11-19         2022-11-12          8                2               
611  2022-11-12         2022-11-12          1                1               
612  2023-01-20         2022-11-12          70               10              
613  2022-11-13         2022-11-13          1                1               

     fan_chat_count  avg_fan_chat  
0    23              7.67          
1    891             89.10         
2    22              3.14          
3    1106            110.60        
4    4               0.80          
..  ..                ...          
609  1               1.00          
610  11              5.50          
611  7               7.00          
612  1988            198.80        
613  5               5.00          

[614 rows x 7 columns]

In [142]:
from datetime import datetime, timedelta

In [169]:
df_fan_date = df_fan.copy()
df_fan_date['time'] = df_fan['time'].apply(lambda x :x.date)

In [171]:
tmp_list = []
for idx, row in df_tmp1.iterrows():
    day = row["last_fan_chat_date"]+timedelta(days=-7)
    room = row["chat_room_id"]
    tmp_list.append( df_fan_date[ (df_fan_date['chat_room_id'] ==room) & (df_fan_date['time'] > day)].shape[0] )
    
tmp_list


[6,
 31,
 1,
 153,
 1,
 8,
 1,
 4,
 125,
 20,
 7,
 7,
 780,
 1,
 4,
 2,
 1,
 3,
 1,
 759,
 88,
 50,
 8,
 28,
 2,
 5,
 1,
 7,
 66,
 5,
 13,
 5,
 707,
 2,
 386,
 28,
 1,
 1,
 49,
 37,
 139,
 5,
 1,
 277,
 404,
 31,
 41,
 868,
 6,
 1,
 1222,
 4,
 170,
 2,
 9,
 21,
 2,
 40,
 51,
 14,
 7,
 125,
 4,
 1040,
 101,
 70,
 123,
 70,
 1,
 583,
 52,
 14,
 24,
 20,
 1123,
 65,
 87,
 108,
 61,
 3,
 231,
 13,
 3,
 120,
 1,
 3,
 26,
 1,
 250,
 1,
 8,
 4,
 2,
 32,
 3,
 5,
 1,
 4,
 5,
 1,
 2,
 14,
 1,
 29,
 34,
 30,
 1,
 1,
 94,
 24,
 2,
 11,
 25,
 1,
 1,
 57,
 4,
 1,
 2,
 48,
 1,
 2,
 1,
 1,
 13,
 1,
 1,
 1,
 1,
 1,
 8,
 1,
 1,
 2,
 2,
 10,
 2,
 13,
 29,
 1,
 1,
 206,
 2,
 1,
 1,
 1,
 1,
 63,
 1,
 18,
 7,
 436,
 1,
 2,
 1668,
 32,
 2,
 59,
 1,
 2,
 4,
 370,
 1,
 1,
 3,
 3,
 1,
 3,
 15,
 2,
 2,
 9,
 1,
 21,
 5,
 8,
 1,
 2,
 1,
 1,
 10,
 161,
 2,
 4,
 1,
 1,
 8,
 1,
 1,
 1,
 68,
 1,
 1,
 192,
 1,
 2,
 46,
 233,
 448,
 37,
 117,
 303,
 2,
 2,
 1,
 1,
 1,
 1,
 96,
 1,
 1,
 137,
 1,
 1,
 1,
 39,
 2,
 1,
 1,


In [174]:
df_tmp1['last_week_fan_chat_count'] = tmp_list
df_tmp1

chat_room_id  \
0    sendbird_group_channel_100098601_8588e70ceb6ad4538f8cd750040503f0281e2256   
1    sendbird_group_channel_100230212_d09bcba2f742f5e2eccfa93929cac768ea781690   
2    sendbird_group_channel_100311155_0b3a8d43c52f47f7fc8a64f7be914c5449f3db9e   
3    sendbird_group_channel_100349582_b3e0ea76ba8e8e82d84c7883fb00249a381b0a09   
4    sendbird_group_channel_100393628_3311cc39eeafde271bc8e3c2c74b6c3b6abb9407   
..                                                                         ...   
609  sendbird_group_channel_99795834_500b186865d605408b22e8774935cb2b7decbd86    
610  sendbird_group_channel_99850650_d7bd96328f34a17ae3e1d754f234f68a17304ce0    
611  sendbird_group_channel_99874745_64724648d72307db2142bc58979ea6e33e09855a    
612  sendbird_group_channel_99875351_3591638d2b87c2551d3521609dd4c3c19fae0d4b    
613  sendbird_group_channel_99954388_1b9a4a36f692e3b4f9b39c0f520e43a498f6f12e    

    last_fan_chat_date first_fan_chat_date  fan_chat_period  fan_chat_week  \
0    2022-11-30         2022-11-15          16               3               
1    2023-01-20         2022-11-15          67               10              
2    2023-01-01         2022-11-16          47               7               
3    2023-01-20         2022-11-17          65               10              
4    2023-01-04         2022-12-06          30               5               
..          ...                ...          ..              ..               
609  2022-11-13         2022-11-13          1                1               
610  2022-11-19         2022-11-12          8                2               
611  2022-11-12         2022-11-12          1                1               
612  2023-01-20         2022-11-12          70               10              
613  2022-11-13         2022-11-13          1                1               

     fan_chat_count  avg_fan_chat  last_week_fan_chat_count  
0    23              7.67          6                         
1    891             89.10         31                        
2    22              3.14          1                         
3    1106            110.60        153                       
4    4               0.80          1                         
..  ..                ...         ..                         
609  1               1.00          1                         
610  11              5.50          1                         
611  7               7.00          7                         
612  1988            198.80        31                        
613  5               5.00          5                         

[614 rows x 8 columns]

### df_sticker에서  땡큐스티커 횟수, 땡큐스티커 평균금액 , 땡큐스티커 누적금액 구하기

In [175]:
df_sticker

chat_room_id  \
0     sendbird_group_channel_51673049_e9127466665f7535749033f986c3e01c7a67cfc1    
1     sendbird_group_channel_82337532_c81ea37759a4761f4f6989e5f0f170f2e7a723fd    
2     sendbird_group_channel_82333993_7fe5b9c11092fdc107a23a6dcb482634eb75b3da    
3     sendbird_group_channel_90863793_15d27a6ad0ea225f4547c61b2b6f7fd58989e4b7    
4     sendbird_group_channel_90683697_dc65fb5afa9b3c6dfae1b4cf4361c433a796d7e5    
...                                                                        ...    
5656  sendbird_group_channel_103089687_a2a01e2acaf8913c5c4df2b096f0d99e38e00e1a   
5657  sendbird_group_channel_82336139_dc97c0b3b177072e91bf628625d5fa7539db532a    
5658  sendbird_group_channel_103089687_a2a01e2acaf8913c5c4df2b096f0d99e38e00e1a   
5659  sendbird_group_channel_95354568_7b070c37a5681beec7577b5f55a1d5c7e7393e89    
5660  sendbird_group_channel_93204183_413a35d15e5c830f0ba87ba75568fd2e44477d6a    

           user_nickname chat_celeb_nickname product_price_string  \
0     밝은달                 🍒체리쉬_문백찬            30                    
1     누리                  🍒체리쉬_반이안            10                    
2     수피아                 🍒체리쉬_윤호림            30                    
3     장미팔                 유 앤🌸                20                    
4     에엘                  예꿍_                 100                   
...   ..                  ...                 ...                   
5656  채리하늘                채티_                 10                    
5657  씨에라                 🍒체리쉬_설 윤            10                    
5658  채리하늘                채티_                 10                    
5659  팬심M 궁물&테스트방💬        써닌장                 10                    
5660  최익디스아웃_나는 곰탱이(정상수)  글라세                 10                    

                          time  
0    2022-09-04 00:10:31+00:00  
1    2022-09-04 00:19:52+00:00  
2    2022-09-04 00:39:44+00:00  
3    2022-09-04 10:18:29+00:00  
4    2022-09-04 13:16:43+00:00  
...                        ...  
5656 2023-01-20 06:51:15+00:00  
5657 2023-01-20 07:52:45+00:00  
5658 2023-01-20 08:17:10+00:00  
5659 2023-01-20 09:32:22+00:00  
5660 2023-01-20 14:32:28+00:00  

[5661 rows x 5 columns]

In [180]:
df_sticker.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5661 entries, 0 to 5660
Data columns (total 5 columns):
 #   Column                Non-Null Count  Dtype              
---  ------                --------------  -----              
 0   chat_room_id          5661 non-null   object             
 1   user_nickname         5511 non-null   object             
 2   chat_celeb_nickname   5661 non-null   object             
 3   product_price_string  5661 non-null   object             
 4   time                  5661 non-null   datetime64[ns, UTC]
dtypes: datetime64[ns, UTC](1), object(4)
memory usage: 221.3+ KB


In [184]:
#문자열을 숫자로 바꿈
df_sticker['price'] = df_sticker['product_price_string'].astype(int)
df_sticker

chat_room_id  \
0     sendbird_group_channel_51673049_e9127466665f7535749033f986c3e01c7a67cfc1    
1     sendbird_group_channel_82337532_c81ea37759a4761f4f6989e5f0f170f2e7a723fd    
2     sendbird_group_channel_82333993_7fe5b9c11092fdc107a23a6dcb482634eb75b3da    
3     sendbird_group_channel_90863793_15d27a6ad0ea225f4547c61b2b6f7fd58989e4b7    
4     sendbird_group_channel_90683697_dc65fb5afa9b3c6dfae1b4cf4361c433a796d7e5    
...                                                                        ...    
5656  sendbird_group_channel_103089687_a2a01e2acaf8913c5c4df2b096f0d99e38e00e1a   
5657  sendbird_group_channel_82336139_dc97c0b3b177072e91bf628625d5fa7539db532a    
5658  sendbird_group_channel_103089687_a2a01e2acaf8913c5c4df2b096f0d99e38e00e1a   
5659  sendbird_group_channel_95354568_7b070c37a5681beec7577b5f55a1d5c7e7393e89    
5660  sendbird_group_channel_93204183_413a35d15e5c830f0ba87ba75568fd2e44477d6a    

           user_nickname chat_celeb_nickname product_price_string  \
0     밝은달                 🍒체리쉬_문백찬            30                    
1     누리                  🍒체리쉬_반이안            10                    
2     수피아                 🍒체리쉬_윤호림            30                    
3     장미팔                 유 앤🌸                20                    
4     에엘                  예꿍_                 100                   
...   ..                  ...                 ...                   
5656  채리하늘                채티_                 10                    
5657  씨에라                 🍒체리쉬_설 윤            10                    
5658  채리하늘                채티_                 10                    
5659  팬심M 궁물&테스트방💬        써닌장                 10                    
5660  최익디스아웃_나는 곰탱이(정상수)  글라세                 10                    

                          time  price  
0    2022-09-04 00:10:31+00:00  30     
1    2022-09-04 00:19:52+00:00  10     
2    2022-09-04 00:39:44+00:00  30     
3    2022-09-04 10:18:29+00:00  20     
4    2022-09-04 13:16:43+00:00  100    
...                        ...  ...    
5656 2023-01-20 06:51:15+00:00  10     
5657 2023-01-20 07:52:45+00:00  10     
5658 2023-01-20 08:17:10+00:00  10     
5659 2023-01-20 09:32:22+00:00  10     
5660 2023-01-20 14:32:28+00:00  10     

[5661 rows x 6 columns]

In [178]:
#df_tmp4 스티커 누적 횟수
df_tmp4 = df_sticker.groupby('chat_room_id',as_index=False).count()[['chat_room_id','time']]
#컬럼이름을 sticker_count로 변경
df_tmp4.rename(columns = {'time':'sticker_count'}, inplace=True)
df_tmp4

chat_room_id  \
0    sendbird_group_channel_100550222_8e976b9673ff62a4c7320b8ff9192b16e30bad95   
1    sendbird_group_channel_100913307_0c46d90c964694d791f535b3a2e76c937ae604f5   
2    sendbird_group_channel_101004369_9e88b897d9a20eca48ba2dc06d38ecd53b9a8bf9   
3    sendbird_group_channel_101172165_5627307dd393998c8530d709e146b8862ac9a82c   
4    sendbird_group_channel_102235779_e2af3a24702cb60dbdcfd6235ebcf430c0185254   
..                                                                         ...   
190  sendbird_group_channel_99486028_cf15a63ef244de447f972e07fb3adc8460c80022    
191  sendbird_group_channel_99533087_8f4447dcba0272bc31a85150f559017a717b06a1    
192  sendbird_group_channel_99575673_b543bd138a7da460543e4a49da513f1614f77781    
193  sendbird_group_channel_99621966_f5fac7c65e4bdacdfaaf39ec116ee02970294321    
194  sendbird_group_channel_99875351_3591638d2b87c2551d3521609dd4c3c19fae0d4b    

     sticker_count  
0    33             
1    1              
2    1              
3    1              
4    7              
..  ..              
190  1              
191  1              
192  4              
193  1              
194  1              

[195 rows x 2 columns]

In [185]:
#df_tmp4 스티커 누적 금액
df_tmp5 = df_sticker.groupby('chat_room_id',as_index=False).sum()[['chat_room_id','price']]
#컬럼이름을 sticker_count로 변경
df_tmp5.rename(columns = {'price':'Total_sticker_sales'}, inplace=True)
df_tmp5

chat_room_id  \
0    sendbird_group_channel_100550222_8e976b9673ff62a4c7320b8ff9192b16e30bad95   
1    sendbird_group_channel_100913307_0c46d90c964694d791f535b3a2e76c937ae604f5   
2    sendbird_group_channel_101004369_9e88b897d9a20eca48ba2dc06d38ecd53b9a8bf9   
3    sendbird_group_channel_101172165_5627307dd393998c8530d709e146b8862ac9a82c   
4    sendbird_group_channel_102235779_e2af3a24702cb60dbdcfd6235ebcf430c0185254   
..                                                                         ...   
190  sendbird_group_channel_99486028_cf15a63ef244de447f972e07fb3adc8460c80022    
191  sendbird_group_channel_99533087_8f4447dcba0272bc31a85150f559017a717b06a1    
192  sendbird_group_channel_99575673_b543bd138a7da460543e4a49da513f1614f77781    
193  sendbird_group_channel_99621966_f5fac7c65e4bdacdfaaf39ec116ee02970294321    
194  sendbird_group_channel_99875351_3591638d2b87c2551d3521609dd4c3c19fae0d4b    

     Total_sticker_sales  
0    350                  
1    10                   
2    20                   
3    10                   
4    350                  
..   ...                  
190  100                  
191  10                   
192  220                  
193  10                   
194  10                   

[195 rows x 2 columns]

In [186]:
df_tmp4 = df_tmp4.merge(df_tmp5, on='chat_room_id')
df_tmp4

chat_room_id  \
0    sendbird_group_channel_100550222_8e976b9673ff62a4c7320b8ff9192b16e30bad95   
1    sendbird_group_channel_100913307_0c46d90c964694d791f535b3a2e76c937ae604f5   
2    sendbird_group_channel_101004369_9e88b897d9a20eca48ba2dc06d38ecd53b9a8bf9   
3    sendbird_group_channel_101172165_5627307dd393998c8530d709e146b8862ac9a82c   
4    sendbird_group_channel_102235779_e2af3a24702cb60dbdcfd6235ebcf430c0185254   
..                                                                         ...   
190  sendbird_group_channel_99486028_cf15a63ef244de447f972e07fb3adc8460c80022    
191  sendbird_group_channel_99533087_8f4447dcba0272bc31a85150f559017a717b06a1    
192  sendbird_group_channel_99575673_b543bd138a7da460543e4a49da513f1614f77781    
193  sendbird_group_channel_99621966_f5fac7c65e4bdacdfaaf39ec116ee02970294321    
194  sendbird_group_channel_99875351_3591638d2b87c2551d3521609dd4c3c19fae0d4b    

     sticker_count  Total_sticker_sales  
0    33             350                  
1    1              10                   
2    1              20                   
3    1              10                   
4    7              350                  
..  ..              ...                  
190  1              100                  
191  1              10                   
192  4              220                  
193  1              10                   
194  1              10                   

[195 rows x 3 columns]

In [189]:
df_tmp4['avg_sticker_price'] = round ( df_tmp4['Total_sticker_sales'] / df_tmp4['sticker_count'] ,1)
df_tmp4

chat_room_id  \
0    sendbird_group_channel_100550222_8e976b9673ff62a4c7320b8ff9192b16e30bad95   
1    sendbird_group_channel_100913307_0c46d90c964694d791f535b3a2e76c937ae604f5   
2    sendbird_group_channel_101004369_9e88b897d9a20eca48ba2dc06d38ecd53b9a8bf9   
3    sendbird_group_channel_101172165_5627307dd393998c8530d709e146b8862ac9a82c   
4    sendbird_group_channel_102235779_e2af3a24702cb60dbdcfd6235ebcf430c0185254   
..                                                                         ...   
190  sendbird_group_channel_99486028_cf15a63ef244de447f972e07fb3adc8460c80022    
191  sendbird_group_channel_99533087_8f4447dcba0272bc31a85150f559017a717b06a1    
192  sendbird_group_channel_99575673_b543bd138a7da460543e4a49da513f1614f77781    
193  sendbird_group_channel_99621966_f5fac7c65e4bdacdfaaf39ec116ee02970294321    
194  sendbird_group_channel_99875351_3591638d2b87c2551d3521609dd4c3c19fae0d4b    

     sticker_count  Total_sticker_sales  avg_sticker_price  
0    33             350                  10.6               
1    1              10                   10.0               
2    1              20                   20.0               
3    1              10                   10.0               
4    7              350                  50.0               
..  ..              ...                   ...               
190  1              100                  100.0              
191  1              10                   10.0               
192  4              220                  55.0               
193  1              10                   10.0               
194  1              10                   10.0               

[195 rows x 4 columns]

In [190]:
df_tmp4.describe()

sticker_count  Total_sticker_sales  avg_sticker_price
count  195.000000     195.000000           195.000000       
mean   29.030769      943.179487           55.745641        
std    97.248004      2465.199503          68.156095        
min    1.000000       10.000000            1.800000         
25%    1.000000       42.000000            16.900000        
50%    3.000000       100.000000           31.700000        
75%    9.000000       560.000000           71.350000        
max    700.000000     15746.000000         505.000000

### 두 임시 데이터프레임 merge

In [221]:
df_fan_sticker = df_tmp1.merge(df_tmp4, how='outer', on = 'chat_room_id')
df_fan_sticker

chat_room_id  \
0    sendbird_group_channel_100098601_8588e70ceb6ad4538f8cd750040503f0281e2256   
1    sendbird_group_channel_100230212_d09bcba2f742f5e2eccfa93929cac768ea781690   
2    sendbird_group_channel_100311155_0b3a8d43c52f47f7fc8a64f7be914c5449f3db9e   
3    sendbird_group_channel_100349582_b3e0ea76ba8e8e82d84c7883fb00249a381b0a09   
4    sendbird_group_channel_100393628_3311cc39eeafde271bc8e3c2c74b6c3b6abb9407   
..                                                                         ...   
628  sendbird_group_channel_93678156_2556341e045bb5e949452ccbd8c052857958d5a2    
629  sendbird_group_channel_95354568_7b070c37a5681beec7577b5f55a1d5c7e7393e89    
630  sendbird_group_channel_95501543_1de3375178cd51ba53364529ca06db4bf54c6332    
631  sendbird_group_channel_97386858_66a252991456ede97cce43dd9933b2d9bf90b32b    
632  sendbird_group_channel_97386859_2935dc8a62ac7b1f6b276b21d7577976e095fa90    

    last_fan_chat_date first_fan_chat_date  fan_chat_period  fan_chat_week  \
0    2022-11-30         2022-11-15          16.0             3.0             
1    2023-01-20         2022-11-15          67.0             10.0            
2    2023-01-01         2022-11-16          47.0             7.0             
3    2023-01-20         2022-11-17          65.0             10.0            
4    2023-01-04         2022-12-06          30.0             5.0             
..          ...                ...           ...             ...             
628  NaN                NaN                NaN              NaN              
629  NaN                NaN                NaN              NaN              
630  NaN                NaN                NaN              NaN              
631  NaN                NaN                NaN              NaN              
632  NaN                NaN                NaN              NaN              

     fan_chat_count  avg_fan_chat  last_week_fan_chat_count  sticker_count  \
0    23.0            7.67          6.0                      NaN              
1    891.0           89.10         31.0                     NaN              
2    22.0            3.14          1.0                      NaN              
3    1106.0          110.60        153.0                    NaN              
4    4.0             0.80          1.0                      NaN              
..   ...              ...          ...                       ..              
628 NaN             NaN           NaN                        3.0             
629 NaN             NaN           NaN                        8.0             
630 NaN             NaN           NaN                        7.0             
631 NaN             NaN           NaN                        17.0            
632 NaN             NaN           NaN                        4.0             

     Total_sticker_sales  avg_sticker_price  
0   NaN                  NaN                 
1   NaN                  NaN                 
2   NaN                  NaN                 
3   NaN                  NaN                 
4   NaN                  NaN                 
..   ..                   ..                 
628  30.0                 10.0               
629  90.0                 11.2               
630  816.0                116.6              
631  180.0                10.6               
632  40.0                 10.0               

[633 rows x 11 columns]

In [222]:
#결측치 확인
df_fan_sticker.isna().sum()

chat_room_id                0  
last_fan_chat_date          19 
first_fan_chat_date         19 
fan_chat_period             19 
fan_chat_week               19 
fan_chat_count              19 
avg_fan_chat                19 
last_week_fan_chat_count    19 
sticker_count               438
Total_sticker_sales         438
avg_sticker_price           438
dtype: int64

팬 채팅기록이 없는데 땡큐스티커 기록만 있는 채팅방이 19개, 채팅기록이 있지만 스티커 기록이 없는방이 438개.  
날짜 이외 컬럼은 결측치를 0으로 대치한다.

In [223]:
df_fan_sticker[['fan_chat_period',	'fan_chat_week'	,'fan_chat_count',	'avg_fan_chat',	'last_week_fan_chat_count',	'sticker_count',	'Total_sticker_sales',	'avg_sticker_price']]= df_fan_sticker[['fan_chat_period',	'fan_chat_week'	,'fan_chat_count',	'avg_fan_chat',	'last_week_fan_chat_count',	'sticker_count',	'Total_sticker_sales',	'avg_sticker_price']].fillna(0)
df_fan_sticker

chat_room_id  \
0    sendbird_group_channel_100098601_8588e70ceb6ad4538f8cd750040503f0281e2256   
1    sendbird_group_channel_100230212_d09bcba2f742f5e2eccfa93929cac768ea781690   
2    sendbird_group_channel_100311155_0b3a8d43c52f47f7fc8a64f7be914c5449f3db9e   
3    sendbird_group_channel_100349582_b3e0ea76ba8e8e82d84c7883fb00249a381b0a09   
4    sendbird_group_channel_100393628_3311cc39eeafde271bc8e3c2c74b6c3b6abb9407   
..                                                                         ...   
628  sendbird_group_channel_93678156_2556341e045bb5e949452ccbd8c052857958d5a2    
629  sendbird_group_channel_95354568_7b070c37a5681beec7577b5f55a1d5c7e7393e89    
630  sendbird_group_channel_95501543_1de3375178cd51ba53364529ca06db4bf54c6332    
631  sendbird_group_channel_97386858_66a252991456ede97cce43dd9933b2d9bf90b32b    
632  sendbird_group_channel_97386859_2935dc8a62ac7b1f6b276b21d7577976e095fa90    

    last_fan_chat_date first_fan_chat_date  fan_chat_period  fan_chat_week  \
0    2022-11-30         2022-11-15          16.0             3.0             
1    2023-01-20         2022-11-15          67.0             10.0            
2    2023-01-01         2022-11-16          47.0             7.0             
3    2023-01-20         2022-11-17          65.0             10.0            
4    2023-01-04         2022-12-06          30.0             5.0             
..          ...                ...           ...             ...             
628  NaN                NaN                 0.0              0.0             
629  NaN                NaN                 0.0              0.0             
630  NaN                NaN                 0.0              0.0             
631  NaN                NaN                 0.0              0.0             
632  NaN                NaN                 0.0              0.0             

     fan_chat_count  avg_fan_chat  last_week_fan_chat_count  sticker_count  \
0    23.0            7.67          6.0                       0.0             
1    891.0           89.10         31.0                      0.0             
2    22.0            3.14          1.0                       0.0             
3    1106.0          110.60        153.0                     0.0             
4    4.0             0.80          1.0                       0.0             
..   ...              ...          ...                       ...             
628  0.0             0.00          0.0                       3.0             
629  0.0             0.00          0.0                       8.0             
630  0.0             0.00          0.0                       7.0             
631  0.0             0.00          0.0                       17.0            
632  0.0             0.00          0.0                       4.0             

     Total_sticker_sales  avg_sticker_price  
0    0.0                  0.0                
1    0.0                  0.0                
2    0.0                  0.0                
3    0.0                  0.0                
4    0.0                  0.0                
..   ...                  ...                
628  30.0                 10.0               
629  90.0                 11.2               
630  816.0                116.6              
631  180.0                10.6               
632  40.0                 10.0               

[633 rows x 11 columns]

In [224]:
#결측치 확인
df_fan_sticker.isna().sum()

chat_room_id                0 
last_fan_chat_date          19
first_fan_chat_date         19
fan_chat_period             0 
fan_chat_week               0 
fan_chat_count              0 
avg_fan_chat                0 
last_week_fan_chat_count    0 
sticker_count               0 
Total_sticker_sales         0 
avg_sticker_price           0 
dtype: int64

In [225]:
df_list

user_nickname  \
0    소나무            
1    아라7            
2    밤양갱            
3    Yewon          
4    mcel01         
..      ...         
878  민지             
879  잇설             
880  견지다            
881  해득이            
882  10810          

                                                                  chat_room_id  
0    sendbird_group_channel_85249736_52d2fa2d3517ffc822f7964bdf82c80e801033cd   
1    sendbird_group_channel_89939017_5d92753dd838235ebee2617f3446a6a8707fc3c5   
2    sendbird_group_channel_95193234_fe13fff1331cae16e336ef3b5968f23e59a3d01c   
3    sendbird_group_channel_92986328_49d9846cdbac6b1d63b58ba560ff895913f9b64e   
4    sendbird_group_channel_66268842_b0e38258ae02c78be067e9abb11ea924d1887994   
..                                                                        ...   
878  sendbird_group_channel_51792327_a4e048a2af07bfd0ac622a91f3eced27b534ce7b   
879  sendbird_group_channel_49477936_9547f900490851dbc81aa00bdc89bd2b43250c96   
880  sendbird_group_channel_99534199_48cbe260c10532e086ccb3a300db337fd4a1ca20   
881  sendbird_group_channel_101060100_19354053a9be596a311c0cef04b3971c1070c780  
882  sendbird_group_channel_106712066_c5f115051ae8d5213ee0d2cf7ded260a7c7c9292  

[883 rows x 2 columns]

채팅방 id를 기준으로 셀럽이름도 merge

In [226]:
df_fan_sticker = df_fan_sticker.merge(df_list, how='left', on='chat_room_id')
df_fan_sticker

chat_room_id  \
0    sendbird_group_channel_100098601_8588e70ceb6ad4538f8cd750040503f0281e2256   
1    sendbird_group_channel_100230212_d09bcba2f742f5e2eccfa93929cac768ea781690   
2    sendbird_group_channel_100311155_0b3a8d43c52f47f7fc8a64f7be914c5449f3db9e   
3    sendbird_group_channel_100349582_b3e0ea76ba8e8e82d84c7883fb00249a381b0a09   
4    sendbird_group_channel_100393628_3311cc39eeafde271bc8e3c2c74b6c3b6abb9407   
..                                                                         ...   
677  sendbird_group_channel_95501543_1de3375178cd51ba53364529ca06db4bf54c6332    
678  sendbird_group_channel_95501543_1de3375178cd51ba53364529ca06db4bf54c6332    
679  sendbird_group_channel_97386858_66a252991456ede97cce43dd9933b2d9bf90b32b    
680  sendbird_group_channel_97386858_66a252991456ede97cce43dd9933b2d9bf90b32b    
681  sendbird_group_channel_97386859_2935dc8a62ac7b1f6b276b21d7577976e095fa90    

    last_fan_chat_date first_fan_chat_date  fan_chat_period  fan_chat_week  \
0    2022-11-30         2022-11-15          16.0             3.0             
1    2023-01-20         2022-11-15          67.0             10.0            
2    2023-01-01         2022-11-16          47.0             7.0             
3    2023-01-20         2022-11-17          65.0             10.0            
4    2023-01-04         2022-12-06          30.0             5.0             
..          ...                ...           ...             ...             
677  NaN                NaN                 0.0              0.0             
678  NaN                NaN                 0.0              0.0             
679  NaN                NaN                 0.0              0.0             
680  NaN                NaN                 0.0              0.0             
681  NaN                NaN                 0.0              0.0             

     fan_chat_count  avg_fan_chat  last_week_fan_chat_count  sticker_count  \
0    23.0            7.67          6.0                       0.0             
1    891.0           89.10         31.0                      0.0             
2    22.0            3.14          1.0                       0.0             
3    1106.0          110.60        153.0                     0.0             
4    4.0             0.80          1.0                       0.0             
..   ...              ...          ...                       ...             
677  0.0             0.00          0.0                       7.0             
678  0.0             0.00          0.0                       7.0             
679  0.0             0.00          0.0                       17.0            
680  0.0             0.00          0.0                       17.0            
681  0.0             0.00          0.0                       4.0             

     Total_sticker_sales  avg_sticker_price user_nickname  
0    0.0                  0.0                팬심M in 지스타    
1    0.0                  0.0                김네로           
2    0.0                  0.0                쫌눈            
3    0.0                  0.0                애츄            
4    0.0                  0.0                유다이           
..   ...                  ...                ...           
677  816.0                116.6              기염둥이          
678  816.0                116.6              우우우우우우주       
679  180.0                10.6               mcel52!!      
680  180.0                10.6               mcel52        
681  40.0                 10.0               mcel53        

[682 rows x 12 columns]

In [210]:
df_fan_sticker[df_fan_sticker.chat_room_id	.duplicated()]

chat_room_id  \
16   sendbird_group_channel_101083581_51e42c2a6374eb7582a2d367c7c27c9045a2dbef   
40   sendbird_group_channel_102749234_433007af4fced2ea1e576ac0ebc9a9342eda8688   
42   sendbird_group_channel_102786836_d9ccd684a8c40adf3b5e103407fe3b9b211e2028   
91   sendbird_group_channel_45236526_b830934347148677440016a5bb37c3360771d618    
92   sendbird_group_channel_45236526_b830934347148677440016a5bb37c3360771d618    
107  sendbird_group_channel_49355617_a7a94e8105e27457bd3492eeb3b38cb49117b6df    
120  sendbird_group_channel_49669010_9e55d83b5b0d92e33d2fe53531aa969cf1bdbea3    
121  sendbird_group_channel_49669010_9e55d83b5b0d92e33d2fe53531aa969cf1bdbea3    
122  sendbird_group_channel_49669010_9e55d83b5b0d92e33d2fe53531aa969cf1bdbea3    
138  sendbird_group_channel_66268842_b0e38258ae02c78be067e9abb11ea924d1887994    
139  sendbird_group_channel_66268842_b0e38258ae02c78be067e9abb11ea924d1887994    
140  sendbird_group_channel_66268842_b0e38258ae02c78be067e9abb11ea924d1887994    
141  sendbird_group_channel_66268842_b0e38258ae02c78be067e9abb11ea924d1887994    
142  sendbird_group_channel_66268842_b0e38258ae02c78be067e9abb11ea924d1887994    
143  sendbird_group_channel_66268842_b0e38258ae02c78be067e9abb11ea924d1887994    
144  sendbird_group_channel_66268842_b0e38258ae02c78be067e9abb11ea924d1887994    
145  sendbird_group_channel_66268842_b0e38258ae02c78be067e9abb11ea924d1887994    
146  sendbird_group_channel_66268842_b0e38258ae02c78be067e9abb11ea924d1887994    
148  sendbird_group_channel_66269958_a541997e47b030dfbe5ff658f2a4447144d782e3    
149  sendbird_group_channel_66269958_a541997e47b030dfbe5ff658f2a4447144d782e3    
150  sendbird_group_channel_66269958_a541997e47b030dfbe5ff658f2a4447144d782e3    
151  sendbird_group_channel_66269958_a541997e47b030dfbe5ff658f2a4447144d782e3    
154  sendbird_group_channel_67153858_1f311d35e429a768f8ffaeffc852c37897dc53f0    
155  sendbird_group_channel_67153858_1f311d35e429a768f8ffaeffc852c37897dc53f0    
232  sendbird_group_channel_85998731_38f9622298b3eb21b41ace4394259cb8dbf7e1ec    
298  sendbird_group_channel_87837725_a9843f05d63c150bd073e6627b28dd1f144e715d    
302  sendbird_group_channel_87971137_5f1a8273a7f2024da2634dc84e6e7e5d2363b15a    
311  sendbird_group_channel_89307945_cd7cae0da88108ed5a37e54950ddc35b400193a1    
313  sendbird_group_channel_89313098_a2d0eccde8e67036e7c2bf8856cc744ff31539ce    
333  sendbird_group_channel_90227072_fd54005c6cfc5d7788c7b762537fd49d54a265f2    
340  sendbird_group_channel_90524524_ce532935677619d3ebd80e2ad33aeff92b270c35    
350  sendbird_group_channel_90863793_15d27a6ad0ea225f4547c61b2b6f7fd58989e4b7    
351  sendbird_group_channel_90863793_15d27a6ad0ea225f4547c61b2b6f7fd58989e4b7    
376  sendbird_group_channel_92251141_65b471b5e5ecbaaab668e1a83d197483e2809133    
410  sendbird_group_channel_93017214_41e7aadd8cc2222a541e10b67e1635ceb579f7e2    
449  sendbird_group_channel_94102728_d6fe0b4f526f17b70a547c934264e942a525b8e5    
450  sendbird_group_channel_94102728_d6fe0b4f526f17b70a547c934264e942a525b8e5    
495  sendbird_group_channel_96630071_1b026b9a373f1cb6a7c88e3589276080a448b160    
555  sendbird_group_channel_106586762_885644acd62565c9acf92a75ffbe7610c098601b   
558  sendbird_group_channel_106706536_0eb613646a20c2c482b11ad4b8f16bf0d42753f8   
559  sendbird_group_channel_106706536_0eb613646a20c2c482b11ad4b8f16bf0d42753f8   
563  sendbird_group_channel_39340896_78826e9dd72a93651e51b10137c1716fc41d34e8    
564  sendbird_group_channel_39340896_78826e9dd72a93651e51b10137c1716fc41d34e8    
565  sendbird_group_channel_39340896_78826e9dd72a93651e51b10137c1716fc41d34e8    
571  sendbird_group_channel_81990476_58db89eff45e1ddb29439cd3183419b67d5fdcbc    
572  sendbird_group_channel_81990476_58db89eff45e1ddb29439cd3183419b67d5fdcbc    
577  sendbird_group_channel_95354568_7b070c37a5681beec7577b5f55a1d5c7e7393e89    
579  sendbird_group_channel_95501543_1de3375178cd51ba53364529ca06db4bf54c6332    
581  sendbird_group_channel_97386858_66a25299145

In [227]:
#채팅방 겹치는것은 마지막것을 살리고 삭제
df_fan_sticker = df_fan_sticker.drop_duplicates(subset='chat_room_id', keep='last')

In [228]:
df_fan_sticker

chat_room_id  \
0    sendbird_group_channel_100098601_8588e70ceb6ad4538f8cd750040503f0281e2256   
1    sendbird_group_channel_100230212_d09bcba2f742f5e2eccfa93929cac768ea781690   
2    sendbird_group_channel_100311155_0b3a8d43c52f47f7fc8a64f7be914c5449f3db9e   
3    sendbird_group_channel_100349582_b3e0ea76ba8e8e82d84c7883fb00249a381b0a09   
4    sendbird_group_channel_100393628_3311cc39eeafde271bc8e3c2c74b6c3b6abb9407   
..                                                                         ...   
674  sendbird_group_channel_93678156_2556341e045bb5e949452ccbd8c052857958d5a2    
676  sendbird_group_channel_95354568_7b070c37a5681beec7577b5f55a1d5c7e7393e89    
678  sendbird_group_channel_95501543_1de3375178cd51ba53364529ca06db4bf54c6332    
680  sendbird_group_channel_97386858_66a252991456ede97cce43dd9933b2d9bf90b32b    
681  sendbird_group_channel_97386859_2935dc8a62ac7b1f6b276b21d7577976e095fa90    

    last_fan_chat_date first_fan_chat_date  fan_chat_period  fan_chat_week  \
0    2022-11-30         2022-11-15          16.0             3.0             
1    2023-01-20         2022-11-15          67.0             10.0            
2    2023-01-01         2022-11-16          47.0             7.0             
3    2023-01-20         2022-11-17          65.0             10.0            
4    2023-01-04         2022-12-06          30.0             5.0             
..          ...                ...           ...             ...             
674  NaN                NaN                 0.0              0.0             
676  NaN                NaN                 0.0              0.0             
678  NaN                NaN                 0.0              0.0             
680  NaN                NaN                 0.0              0.0             
681  NaN                NaN                 0.0              0.0             

     fan_chat_count  avg_fan_chat  last_week_fan_chat_count  sticker_count  \
0    23.0            7.67          6.0                       0.0             
1    891.0           89.10         31.0                      0.0             
2    22.0            3.14          1.0                       0.0             
3    1106.0          110.60        153.0                     0.0             
4    4.0             0.80          1.0                       0.0             
..   ...              ...          ...                       ...             
674  0.0             0.00          0.0                       3.0             
676  0.0             0.00          0.0                       8.0             
678  0.0             0.00          0.0                       7.0             
680  0.0             0.00          0.0                       17.0            
681  0.0             0.00          0.0                       4.0             

     Total_sticker_sales  avg_sticker_price user_nickname  
0    0.0                  0.0                팬심M in 지스타    
1    0.0                  0.0                김네로           
2    0.0                  0.0                쫌눈            
3    0.0                  0.0                애츄            
4    0.0                  0.0                유다이           
..   ...                  ...                ...           
674  30.0                 10.0               mcel40        
676  90.0                 11.2               mcel510       
678  816.0                116.6              우우우우우우주       
680  180.0                10.6               mcel52        
681  40.0                 10.0               mcel53        

[633 rows x 12 columns]

In [229]:
#결측치 다시 확인
df_fan_sticker.isna().sum()

chat_room_id                0 
last_fan_chat_date          19
first_fan_chat_date         19
fan_chat_period             0 
fan_chat_week               0 
fan_chat_count              0 
avg_fan_chat                0 
last_week_fan_chat_count    0 
sticker_count               0 
Total_sticker_sales         0 
avg_sticker_price           0 
user_nickname               99
dtype: int64

In [241]:
df_tmp7 =df_fan_sticker[df_fan_sticker['user_nickname'].isna()]
df_tmp7

chat_room_id  \
98   sendbird_group_channel_41626803_28fe1747e2ba0e6c129b4b91832761467b0a4598   
99   sendbird_group_channel_42176637_bdbeb129272deb568d36c79b99bb6fcca0fb6782   
107  sendbird_group_channel_46928750_a7e99249b2549424100b4ca1cfaac4a7ad2e579c   
111  sendbird_group_channel_47177928_6d5a8808975a44b7b6f6864e25e4785aff4120c6   
112  sendbird_group_channel_47285998_c55544ca7673a3c98e754cd8aec6d69abd2dc220   
..                                                                        ...   
617  sendbird_group_channel_98165030_54ceb4a602f8c86134754234d3564cc399bc94ab   
618  sendbird_group_channel_98209515_8cc72b323985efb8af521e0b11b24b3c2072a768   
630  sendbird_group_channel_99288220_d1ce1f3ac38e8f4747ced9d22a860f613bc3dc9c   
645  sendbird_group_channel_99700178_fd9ce134b64cab5e815c5672c6453d53e6d2bbbe   
667  sendbird_group_channel_77763481_f77393f8438556d85659442b4aec9163236f6715   

    last_fan_chat_date first_fan_chat_date  fan_chat_period  fan_chat_week  \
98   2022-09-09         2022-09-07          3.0              1.0             
99   2022-11-24         2022-11-24          1.0              1.0             
107  2022-11-24         2022-09-09          77.0             11.0            
111  2022-09-21         2022-09-21          1.0              1.0             
112  2022-09-22         2022-09-22          1.0              1.0             
..          ...                ...          ...              ...             
617  2022-10-29         2022-10-29          1.0              1.0             
618  2022-11-16         2022-11-16          1.0              1.0             
630  2023-01-08         2023-01-08          1.0              1.0             
645  2022-12-07         2022-12-07          1.0              1.0             
667  NaN                NaN                 0.0              0.0             

     fan_chat_count  avg_fan_chat  last_week_fan_chat_count  sticker_count  \
98   5.0             5.00          5.0                       0.0             
99   1.0             1.00          1.0                       0.0             
107  4.0             0.36          1.0                       0.0             
111  1.0             1.00          1.0                       0.0             
112  1.0             1.00          1.0                       0.0             
..   ...              ...          ...                       ...             
617  1.0             1.00          1.0                       0.0             
618  1.0             1.00          1.0                       0.0             
630  1.0             1.00          1.0                       0.0             
645  1.0             1.00          1.0                       0.0             
667  0.0             0.00          0.0                       1.0             

     Total_sticker_sales  avg_sticker_price user_nickname  
98   0.0                  0.0                NaN           
99   0.0                  0.0                NaN           
107  0.0                  0.0                NaN           
111  0.0                  0.0                NaN           
112  0.0                  0.0                NaN           
..   ...                  ...                ...           
617  0.0                  0.0                NaN           
618  0.0                  0.0                NaN           
630  0.0                  0.0                NaN           
645  0.0                  0.0                NaN           
667  100.0                100.0              NaN           

[91 rows x 12 columns]

In [258]:
df_access[ df_access['chat_room_id'] == 'sendbird_group_channel_98165030_54ceb4a602f8c86134754234d3564cc399bc94ab']['user_nickname'].mode()[0]

'청컨'

In [260]:
#df_access 데이터프레임에서 채팅방 id로 셀럽이름 찾아서 결측치 메꾸기

for idx,row in df_tmp7.iterrows():
    if len(df_access[df_access['chat_room_id'] == row['chat_room_id']]):
        tmp_name = df_access[df_access['chat_room_id'] == row['chat_room_id']]['user_nickname'].mode()[0]
        df_fan_sticker.loc[idx,'user_nickname'] = tmp_name

In [261]:
#결측치 다시 확인
df_fan_sticker.isna().sum()

chat_room_id                0 
last_fan_chat_date          19
first_fan_chat_date         19
fan_chat_period             0 
fan_chat_week               0 
fan_chat_count              0 
avg_fan_chat                0 
last_week_fan_chat_count    0 
sticker_count               0 
Total_sticker_sales         0 
avg_sticker_price           0 
user_nickname               50
dtype: int64

그래도 결측치 50개가 남아있다. 셀럽 이름을 알 수 없는 채팅방 데이터를 어떻게 할지는 추후 결정하기로 함

In [262]:
df_fan_sticker

chat_room_id  \
0    sendbird_group_channel_100098601_8588e70ceb6ad4538f8cd750040503f0281e2256   
1    sendbird_group_channel_100230212_d09bcba2f742f5e2eccfa93929cac768ea781690   
2    sendbird_group_channel_100311155_0b3a8d43c52f47f7fc8a64f7be914c5449f3db9e   
3    sendbird_group_channel_100349582_b3e0ea76ba8e8e82d84c7883fb00249a381b0a09   
4    sendbird_group_channel_100393628_3311cc39eeafde271bc8e3c2c74b6c3b6abb9407   
..                                                                         ...   
674  sendbird_group_channel_93678156_2556341e045bb5e949452ccbd8c052857958d5a2    
676  sendbird_group_channel_95354568_7b070c37a5681beec7577b5f55a1d5c7e7393e89    
678  sendbird_group_channel_95501543_1de3375178cd51ba53364529ca06db4bf54c6332    
680  sendbird_group_channel_97386858_66a252991456ede97cce43dd9933b2d9bf90b32b    
681  sendbird_group_channel_97386859_2935dc8a62ac7b1f6b276b21d7577976e095fa90    

    last_fan_chat_date first_fan_chat_date  fan_chat_period  fan_chat_week  \
0    2022-11-30         2022-11-15          16.0             3.0             
1    2023-01-20         2022-11-15          67.0             10.0            
2    2023-01-01         2022-11-16          47.0             7.0             
3    2023-01-20         2022-11-17          65.0             10.0            
4    2023-01-04         2022-12-06          30.0             5.0             
..          ...                ...           ...             ...             
674  NaN                NaN                 0.0              0.0             
676  NaN                NaN                 0.0              0.0             
678  NaN                NaN                 0.0              0.0             
680  NaN                NaN                 0.0              0.0             
681  NaN                NaN                 0.0              0.0             

     fan_chat_count  avg_fan_chat  last_week_fan_chat_count  sticker_count  \
0    23.0            7.67          6.0                       0.0             
1    891.0           89.10         31.0                      0.0             
2    22.0            3.14          1.0                       0.0             
3    1106.0          110.60        153.0                     0.0             
4    4.0             0.80          1.0                       0.0             
..   ...              ...          ...                       ...             
674  0.0             0.00          0.0                       3.0             
676  0.0             0.00          0.0                       8.0             
678  0.0             0.00          0.0                       7.0             
680  0.0             0.00          0.0                       17.0            
681  0.0             0.00          0.0                       4.0             

     Total_sticker_sales  avg_sticker_price user_nickname  
0    0.0                  0.0                팬심M in 지스타    
1    0.0                  0.0                김네로           
2    0.0                  0.0                쫌눈            
3    0.0                  0.0                애츄            
4    0.0                  0.0                유다이           
..   ...                  ...                ...           
674  30.0                 10.0               mcel40        
676  90.0                 11.2               mcel510       
678  816.0                116.6              우우우우우우주       
680  180.0                10.6               mcel52        
681  40.0                 10.0               mcel53        

[633 rows x 12 columns]

In [263]:
df_fan_sticker.to_csv('cp2_팬채팅_스티커_데이터.csv', index=False)